collections/src/test/kotlin/ktx/collections/arraysTest.kt

```diff
@@ -352,6 +354,50 @@ class ArraysTest {
   }
 
   @Test
+  fun `should map elements into a new GdxArray`() {
+    val array = GdxArray.with(1, 2, 3)
+    val result = array.map { it * 2 }
+
+    assertTrue(result is GdxArray)
+    assertEquals(2, result[0])
+    assertEquals(4, result[1])
+    assertEquals(6, result[2])
+  }
+
+  @Test
+  fun `should filter elements into a new GdxArray`() {
+    val array = GdxArray.with(1, 2, 3, 4, 5)
+    val result = array.filter { it % 2 == 1 }
+
+    assertTrue(result is GdxArray)
+    assertEquals(3, result.size)
+    assertEquals(1, result[0])
+    assertEquals(3, result[1])
+    assertEquals(5, result[2])
+  }
+
+  @Test
+  fun `should flatten elements into a new GdxArray`() {
+    val array = GdxArray.with(GdxArray.with(1), listOf<Int>(), LinkedList(arrayListOf(2, 3)))
+    val result = array.flatten()
+
+    assertTrue(result is GdxArray)
+    assertEquals(3, result.size)
+    assertEquals(1, result[0])
+    assertEquals(2, result[1])
+    assertEquals(3, result[2])
+  }
+
+  @Test
+  fun `should map elements to lists and flatten them into a new GdxArray`() {
+    val array = GdxArray.with(1, 2, 3)
+    val result = array.flatMap { List(it) { "" }  }
```

Comment:
This is not the best test of `flatMap`, as this does not ensure that each element was transformed. You can use `it` inside `List(it)` body and check if the result is equal to `1, 2, 2, 3, 3, 3`.